### ISSUE TITLE
[beta] debug macros work incorrectly in 3.13.0-beta.3

### ISSUE THREAD

**Issue by lennyburdette** [emberjs/ember.js#18294]
Reproduction:

```sh
npx ember-cli new macro-test # will install 3.12
cd macro-test
$(npm bin)/ember build --environment production
grep "could not find a suitable method to bind" dist/assets/vendor-*.js
# note there's no output

npm install --save ember-source@3.13.0-beta.3
$(npm bin)/ember build --environment production
grep "could not find a suitable method to bind" dist/assets/vendor-*.js
# this will find the assert
```
[The assert in question](https://github.com/emberjs/ember.js/blob/v3.13.0-beta.3/packages/%40ember/runloop/index.js#L226) should be stripped from the build.

For whatever reason, leaving this assert in causes my app to fail to load in production, even though this test app works just fine.

cc @rwjblue

**Comment by rwjblue** [comment]
Reviewing the diff between production builds of 3.12 vs 3.13.0-beta.3 shows a few major differences:

* no longer using `enifed` (we apparently swapped to using `define` again)
* all debug statements are **not** stripped
* the build size is ~ 30k larger (after min+gzip)

Definitely a pretty bad regression, thank you for reporting!

**Comment by rwjblue** [comment]
Fixed by #18304 and also pulled that change into the beta branch (for release in 3.13.0-beta.4 and also the general ember-beta ember-try scenarios).

**Comment by rwjblue** [comment]
Thanks again for reporting @lennyburdette!
